UICatalog/Scenarios/Snake.cs

```diff
@@ -0,0 +1,351 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Terminal.Gui;
+using Terminal.Gui.Graphs;
+using Attribute = Terminal.Gui.Attribute;
+
+namespace UICatalog.Scenarios {
+	[ScenarioMetadata (Name: "Snake", Description: "The game of apple eating.")]
+	[ScenarioCategory ("Colors")]
+	public class Snake : Scenario {
+		private bool isDisposed;
```

Comment:
I didn't notice this in Animation until after I merged but I don't think this works right. In one run i got an exception in the `while(!isDisposed)` loop because I think I hit exit at just the right time. Are you sure this is the proper way to detect the object has been deleted?

Reply:
Hmn yeah... looks like `Scenario` is disposable (`class Scenario : IDisposable`) but nobody actually ever disposes it.  Should be an easy fix.  Also need to update it to the latest line canvas signature.